### DISCUSSION TITLE
expose actions over `niri msg`

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#162] [Ideas]
hey there,
i'm happy to see `niri msg` added as a way to communicate with niri.
i was thinking, it would be nice to see actions exposed thru such a cli as well.
this would for example facilitate scripting niri actions thru the likes of `fuzzel`,
to enable more mouse-based means of interacting with the system. :)

**Comment by YaLTeR** [comment]
Sounds reasonable I guess, what do you have in mind?

**Reply by KiaraGrouwstra** [reply]
my first use-case was porting a logout action (i.e. `exit`).
i imagine one might think of more in a similar vein - perhaps others could chime in there on what might make sense.
in this case i'm porting from an invocation `hyprctl dispatch exit`, and i guess [their approach](https://wiki.hyprland.org/Configuring/Using-hyprctl/#dispatch) of exposing actions similar to how they can be used from keybinding configurations could make sense.

**Reply by YaLTeR** [reply]
Yeah, I think syntax like `niri msg action move-column-to-workspace-up` would makes sense.

**Reply by YaLTeR** [reply]
Implemented in 93e16a6582f8e80ac4079ac527637861696ed7b1.
